### PR TITLE
Find sitemap by sitedefinition

### PR DESCRIPTION
**Pull request by patkleef**

### CODE REVIEW

File: src/Geta.SEO.Sitemaps/Repositories/SitemapRepository.cs [hunk 1]
```diff
@@ -48,7 +54,12 @@ public SitemapData GetSitemapData(string requestUrl)
             
             var host = url.Path.TrimStart('/').ToLowerInvariant();
 
-            return GetAllSitemapData().FirstOrDefault(x => GetHostWithLanguage(x) == host && (x.SiteUrl == null || x.SiteUrl.Contains(url.Host)));
+            var siteDefinition = _siteDefinitionResolver.GetByHostname(url.Host, true, out _);
+            if (siteDefinition == null)
+            {
+                return null;
+            }
+            return GetAllSitemapData().FirstOrDefault(x => GetHostWithLanguage(x) == host && (x.SiteUrl == null || siteDefinition.Hosts.Any(h => h.Name == new Url(x.SiteUrl).Host)));
```
Comment (brianweet) [inline src/Geta.SEO.Sitemaps/Repositories/SitemapRepository.cs:62]: Can you split this up? It's hard to read and the line is too long

File: src/Geta.SEO.Sitemaps/Repositories/SitemapRepository.cs [hunk 2]
```diff
@@ -17,11 +18,16 @@ namespace Geta.SEO.Sitemaps.Repositories
...
     {
         private readonly ILanguageBranchRepository _languageBranchRepository;
+        private readonly ISiteDefinitionResolver _siteDefinitionResolver;
 
-        public SitemapRepository(ILanguageBranchRepository languageBranchRepository)
+
+        public SitemapRepository(ILanguageBranchRepository languageBranchRepository, ISiteDefinitionResolver siteDefinitionResolver)
         {
             if (languageBranchRepository == null) throw new ArgumentNullException("languageBranchRepository");
+            if (siteDefinitionResolver == null) throw new ArgumentNullException("siteDefinitionResolver");
```
Comment (brianweet) [inline src/Geta.SEO.Sitemaps/Repositories/SitemapRepository.cs:27]: Use `nameof(siteDefinitionResolver)` instead, same for the existing param

File: src/Geta.SEO.Sitemaps/Repositories/SitemapRepository.cs (hunk 1 above)
Comment (patkleef) [reply src/Geta.SEO.Sitemaps/Repositories/SitemapRepository.cs]: Added unit tests so we easily test the functionality

File: test/Geta.SEO.Sitemaps.Tests/SitemapRepositoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            
+            var sitemapDataList = new List<SitemapData>
+            {
+                expectedSitemapData
+            };
+
+            var sitemapLoader = new Mock<ISitemapLoader>();
+            sitemapLoader
+                .Setup(x => x.GetAllSitemapData())
+                .Returns(sitemapDataList);
```
Comment (brianweet) [inline test/Geta.SEO.Sitemaps.Tests/SitemapRepositoryTests.cs:161]: Shouldn't this list should contain two sitemaps: one for `xyz.com` and one for `abc.xyz.com` ?

Comment (brianweet) [reply test/Geta.SEO.Sitemaps.Tests/SitemapRepositoryTests.cs:161]: And have to check if the order in the list matters, perhaps add two unit tests?

File: test/Geta.SEO.Sitemaps.Tests/SitemapRepositoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+            var siteMapService = new SitemapRepository(_languageBranchRepository.Object, siteDefinitionResolver.Object, sitemapLoader.Object);
+
+            var siteMapData = siteMapService.GetSitemapData(requestUrl);
+
+            Assert.True(siteMapData.Equals(expectedSitemapData));
```
[the remark this answers is not in the document]
Comment (patkleef) [reply test/Geta.SEO.Sitemaps.Tests/SitemapRepositoryTests.cs]: Done

File: test/Geta.SEO.Sitemaps.Tests/SitemapRepositoryTests.cs (hunk 3 above)
Comment (patkleef) [reply test/Geta.SEO.Sitemaps.Tests/SitemapRepositoryTests.cs:167]: Using the `InlineData` attribute to test several scenarios

**Pull request merged**